### PR TITLE
[Website] content clarification; a11y updates

### PR DESCRIPTION
**Pull request by riley**
connects #427 

### Description
- expand on RTL acronyms
- add Example h2 for a11y reasons
- change first example title to be descriptive
- link-ify Icon category titles
- add to first Link description to explain why there's a discrete Link component at all
- add a bit of explanation about title vs alt attributes on an Icon
- fix some broken links

### Motivation and context
tightening site content and improving a11y.

https://expand-rtl--mineral-ui.netlify.com/

### How to test

1. `cd mineral-ui && npm start`
2. `npm run build:icons`
3. Navigate to each component page. There should be an H2 "Examples" followed by a demo with a more descriptive title.
4. Icon category links should be available
5. does the copy on the Icon `title` attribute example make sense?
6. does the copy on the first Link example make sense?

### Types of changes
- docs updates

### Checklist

* [x] Renders and functions properly in [all supported browsers](https://github.com/mineral-ui/mineral-ui#browser-support)
* [x] Automated tests written and passing [n/a]
* [x] [Accessibility](http://webaim.org/intro) and [inclusivity](https://24ways.org/2016/what-the-heck-is-inclusive-design/) considered
* [x] Rendering performance (initial load time & 60fps) and [perceived performance](http://blog.teamtreehouse.com/perceived-performance) considered [n/a]
* [x] Documentation created or updated
* [x] Tested in [sandbox](https://github.com/facebookincubator/create-react-app#getting-started) if new component or breaking change [n/a]

### How does this PR make you feel?
time to clean up

### CODE REVIEW

File: src/website/app/demos/Button/examples/rtl.js [hunk 1]
```diff
@@ -21,8 +21,9 @@ import IconBackspace from '../../../../../Icon/IconBackspace';
 
 export default {
   id: 'rtl',
-  title: 'RTL Support',
-  description: `Buttons with Icons are reversed when the \`rtl\` prop is set.
+  title: 'Bidirectionality',
+  description: `Buttons support right-to-left (RTL) languages.
+Buttons with Icons are reversed when the \`direction\` prop is set to \`rtl\`.
```
Comment (kylegach) [inline src/website/app/demos/Button/examples/rtl.js]: Button doesn't have a `direction` prop. Did you mean the theme variable?

File: src/website/app/demos/Card/examples/card-image/withImage.js [hunk 2]
```diff
@@ -21,7 +21,7 @@ import DemoLayout from '../../components/DemoLayout';
 
 export default {
   id: 'with-image',
-  title: 'Example',
+  title: 'Displaying Images in a Card',
```
Comment (kylegach) [inline src/website/app/demos/Card/examples/card-image/withImage.js]: It's doubtful they'd need to display more than one image in a Card. Should this be "Displaying an Image in a Card" instead?

File: src/website/app/demos/Icon/examples/title.js [hunk 3]
```diff
@@ -20,8 +20,9 @@ import IconSentimentSatisfied from '../../../../../Icon/IconSentimentSatisfied';
 export default {
   id: 'title',
   title: 'Title',
-  description:
-    'Standalone icons need a title, as opposed to decorative icons (those that repeat the information conveyed by text or do not add significant value).',
+  description: `Standalone icons need a title, as opposed to decorative icons (those that repeat the information conveyed by text or do not add significant value).
+Mineral UI generates the hooks for assistive technology under the covers after you set the \`title\` attribute here.
```
Comment (kylegach) [inline src/website/app/demos/Icon/examples/title.js]: "under the covers" seems off-tone for us.

"...generates the code for assistive technology after you..." instead?

File: src/website/app/demos/Menu/examples/menu-item/rtl.js [hunk 4]
```diff
@@ -22,10 +22,10 @@ import DemoLayout from '../../components/DemoLayout';
 
 export default {
   id: 'rtl',
-  title: 'RTL Support',
+  title: 'Bidirectionality',
   // $FlowFixMe
   backgroundColor: mineralTheme.color_gray_10,
-  description: `MenuItems with Icons are reversed when the \`rtl\` prop is set.
+  description: `MenuItems with Icons are reversed when the \`direction\` prop is set to \`rtl\` (right-to-left).
```
Comment (kylegach) [inline src/website/app/demos/Menu/examples/menu-item/rtl.js]: Not a prop, a theme variable.

File: src/website/app/demos/Card/examples/card-image/withImage.js (hunk 2 above)
Comment (riley) [reply src/website/app/demos/Card/examples/card-image/withImage.js]: 👍

File: src/website/app/demos/Button/examples/rtl.js (hunk 1 above)
Comment (riley) [reply src/website/app/demos/Button/examples/rtl.js]: whoa, changed

File: src/website/app/demos/Icon/examples/title.js (hunk 3 above)
Comment (riley) [reply src/website/app/demos/Icon/examples/title.js]: this is a fair assesment

**Pull request merged**